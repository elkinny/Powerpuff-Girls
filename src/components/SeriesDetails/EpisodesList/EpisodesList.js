import React from 'react';
import { useSelector } from 'react-redux';

import { episodesSelector } from '../../../store/selectors';
import Episode from './Episode';
import style from './style.module.scss';

const EpisodesList = () => {
  const episodes = useSelector(episodesSelector);

  return (
    <div className={style.list}>
      {episodes.map(
        (el) => (
          <Episode
            key={`${el.season}-${el.number}`}
            id={el.id}
            name={el.name}
            season={el.season}
            number={el.number}
            image={el.image ? el.image.original : '/placeholder.png'}
          />
        ),
      )}
    </div>
  );
};

export default EpisodesList;
