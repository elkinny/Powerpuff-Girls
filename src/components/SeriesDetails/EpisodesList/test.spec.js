import React from 'react';
import { render } from '@testing-library/react';
import * as reactRedux from 'react-redux';

import EpisodesList from './EpisodesList';

jest.mock('./Episode', () => (props) => `Episode: ${JSON.stringify(props)}`);

describe('EpisodesList', () => {
  beforeAll(() => {
    jest.spyOn(reactRedux, 'useSelector').mockImplementation();
  });

  test('renders EpisodesList component', () => {
    jest
      .spyOn(reactRedux, 'useSelector')
      .mockReturnValue([
        { seson: 1, number: 1, image: { original: '/img.png' } },
        { seson: 1, number: 2, image: { original: '/img.png' } },
      ]);
    const { container } = render(<EpisodesList />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders EpisodesList component without passed image url', () => {
    jest
      .spyOn(reactRedux, 'useSelector')
      .mockReturnValue([{ seson: 1, number: 1 }, { seson: 1, number: 2 }]);
    const { container } = render(<EpisodesList />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
