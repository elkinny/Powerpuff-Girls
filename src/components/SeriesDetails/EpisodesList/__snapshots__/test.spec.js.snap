// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesList renders EpisodesList component 1`] = `
<div>
  Episode: {"number":1,"image":"/img.png"}
  Episode: {"number":2,"image":"/img.png"}
</div>
`;

exports[`EpisodesList renders EpisodesList component without passed image url 1`] = `
<div>
  Episode: {"number":1,"image":"/placeholder.png"}
  Episode: {"number":2,"image":"/placeholder.png"}
</div>
`;
